specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json

```diff
@@ -1188,6 +1431,137 @@
         }
       }
     },
+    "TaskRunProperties": {
+      "description": "The properties of task run.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "description": "The provisioning state of this task run",
+          "enum": [
+            "Creating",
+            "Updating",
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Canceled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "runRequest": {
+            "$ref": "#/definitions/RunRequest",
+            "description": "The request (parameters) for the run"
+        },
+        "runResult": {
+            "$ref": "#/definitions/Run",
+            "description": "The result of this task run",
+            "readOnly": true
+        }  
+      }
+    },
+    "UserIdentityProperties": {
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "description": "The principal id of user assigned identity.",
+          "type": "string"
+        },
+        "clientId": {
+          "description": "The client id of user assigned identity.",
+          "type": "string"
+        }
+      }
+    },
+    "TaskRunUpdateParameters": {
+      "description": "The parameters for updating a task run.",
+      "type": "object",
+      "properties": {
+        "identity": {
+          "$ref": "#/definitions/IdentityProperties",
+          "description": "Identity for the resource."
+        },
+        "properties": {
+          "$ref": "#/definitions/TaskRunPropertiesUpdateParameters",
+          "description": "The properties for updating a task run.",
+          "x-ms-client-flatten": true
+        },
+        "tags": {
+          "description": "The ARM resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "TaskRunPropertiesUpdateParameters": {
+      "description": "The properties of a task run update parameters.",
+      "type": "object",
+      "properties": {
+        "runRequest": {
+          "$ref": "#/definitions/RunRequest",
```

Comment:
Kristopher, we have been using string for type for several versions. Since it's not related to this PR, are you OK we investigate a little bit more and start another PR to address it?

---
In reply to: [324002726](https://github.com/Azure/azure-rest-api-specs/pull/7123#discussion_r324002726) [](ancestors = 324002726)